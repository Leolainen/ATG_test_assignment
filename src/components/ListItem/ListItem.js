import React from "react";
import PropTypes from "prop-types";
import classnames from "classnames";
import styles from "./styles.module.css";

const ListItem = (props) => {
  const {
    className,
    children,
    component: Component = "li",
    disableDivider = false,
    disablePadding = false,
    ...other
  } = props;

  return (
    <Component
      className={classnames(
        styles.root,
        {
          [styles.padding]: !disablePadding,
          [styles.divider]: !disableDivider,
        },
        className
      )}
      {...other}
    >
      {children}
    </Component>
  );
};

ListItem.propTypes = {
  children: PropTypes.node.isRequired,
  className: PropTypes.string,
};

ListItem.uiName = "ListItem";

export default ListItem;
