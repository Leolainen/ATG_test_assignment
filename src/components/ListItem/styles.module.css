.root {
  text-decoration: none;
  color: var(--harmony-1);
}

.padding {
  padding: 8px 16px;
}

.divider {
  border-bottom: 1px solid var(--harmony-5);
  background-clip: border-box;
}
