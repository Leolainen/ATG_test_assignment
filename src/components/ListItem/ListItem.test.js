import React from "react";
import { render, cleanup } from "@testing-library/react";
import ListItem from "./ListItem";

describe("<ListItem />", () => {
  afterEach(cleanup);

  describe("Snapshots", () => {
    it("should render", () => {
      const { asFragment } = render(<ListItem>list item</ListItem>);

      expect(asFragment()).toMatchSnapshot();
    });

    it("should render without padding and divider", () => {
      const { asFragment } = render(
        <ListItem disablePadding disableDivider>
          list item
        </ListItem>
      );

      expect(asFragment()).toMatchSnapshot();
    });
  });
});
