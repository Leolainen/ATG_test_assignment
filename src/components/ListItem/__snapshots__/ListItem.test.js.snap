// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ListItem /> Snapshots should render 1`] = `
<DocumentFragment>
  <li
    class="root padding divider"
  >
    list item
  </li>
</DocumentFragment>
`;

exports[`<ListItem /> Snapshots should render without padding and divider 1`] = `
<DocumentFragment>
  <li
    class="root"
  >
    list item
  </li>
</DocumentFragment>
`;
